sql/rpl_gtid.cc

```diff
@@ -3525,6 +3526,32 @@ my_bool Window_gtid_event_filter::has_finished()
   return m_has_stop ? m_has_passed : FALSE;
 }
 
+my_bool Window_gtid_event_filter::verify_completed_state()
+{
+  /*TODO: validate `--start-position` (::m_start)
+    Normally with ::m_has_start, the states goes from inactive to ::m_is_active,
+    then to ::m_has_passed if ::m_has_stop (::has_finished()).
+    However, the filter can finish at ::m_has_passed without ever becoming
+    ::m_is_active if ::is_range_invalid(), complicating checks.
+    Ideally, ::is_range_invalid() should warn outside of
+    `--gtid-strict-mode` as well, but that's its own distinct discussion.
+  */
+  if (m_has_stop && !m_has_passed)
+  {
+    /*XXX:
+      `mysqlbinlog.cc`'s `static warning()` family is all duplicate
+      code of the Binlog_gtid_state_validator::warn() family, except
+      it specifies stderr and has a `fflush(result_file)` step.
+    */
+    Binlog_gtid_state_validator::warn(stderr,
+      "Did not reach stop position %u-%u-%llu before end of input",
```

Comment:
`%u-%u-%llu` is used throughout this file (and likely more); but GTID struct fields are actually fixed-size, whereas these formats varies by platform.
I’ll file this mismatch as a separate subissue of https://github.com/MariaDB/server/pull/3360#pullrequestreview-2421203312 ([MDEV-21978](https://jira.mariadb.org/browse/MDEV-21978)).

Reply:
[MDEV-35451](https://jira.mariadb.org/browse/MDEV-35451)